src/progress.rs

```diff
@@ -230,6 +244,85 @@ impl ProgressSet {
             self.progress.len()
         );
     }
+
+    /// Returns the minimal committed index for the cluster.
+    ///
+    /// Eg. If the matched indexes are [2,2,2,4,5], it will return 2.
+    pub fn minimum_committed_index(&self) -> u64 {
+        let mut matched = self
+            .voters()
+            .map(|(_id, peer)| peer.matched)
+            .collect::<Vec<_>>();
```

Comment:
According to the old implement, the alloc here could be an issue. So how about let the `ProgressSet` hold a buffer to sort them?

Reply:
+1

Reply:
Oh, cool idea! I'll do this.